tla/validate.sh

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env bash
+
+WORKDIR="$(mktemp -d)"
+TOOLDIR="${WORKDIR}/tool"
+STATEDIR="${WORKDIR}/state"
+WORKDIR="$(mktemp -d)"
+TOOLDIR="${WORKDIR}/tool"
+STATEDIR="${WORKDIR}/state"
+FAILFAST=false
+PARALLEL=$(nproc)
+
+function show_usage {
+    echo "usage: validate.sh [-p <parallel>] -s <spec> -c <config> <trace files>">&2
+}
+
+function install_tlaplus {
+    echo -n "Downloading TLA+ tools ... "
+    wget -qN https://nightly.tlapl.us/dist/tla2tools.jar -P ${TOOLDIR}
+    wget -qN https://github.com/tlaplus/CommunityModules/releases/latest/download/CommunityModules-deps.jar -P ${TOOLDIR}
+    echo "done."
+}
+PARALLEL=$(nproc)
+
+function preprocess_log {
+    local trace=${1}
+    sed -i -E 's/^[^{]+//' ${trace}
+    sort -t":" -k 3 ${trace} -o ${trace}
+}
+
+function colored_text {
+    color=$1
+    text=$2
+
+    case $color in
+        "red")
+            echo $"\033[0;31m$text\033[0m"
+        ;;
+
+        "green")
+            echo "\033[0;32m$text\033[0m"
+        ;;
+
+        *)
+            echo text
+        ;;
+    esac
+}
+
+function validate {
+    local id=${1}
+    local trace=${2}
+    local spec=${3}
+    local config=${4}
+    local tooldir=${5}
+    local statedir=${6}
+    local name=$(basename $trace .ndjson)
+    local id=${1}
+    local trace=${2}
+    local spec=${3}
+    local config=${4}
+    local tooldir=${5}
+    local statedir=${6}
+    local name=$(basename $trace .ndjson)
+
+    # preprocess log
+    sed -i -E 's/^[^{]+//' ${trace}
+    sort -t":" -k 3 ${trace} -o ${trace}
+    # preprocess log
+    sed -i -E 's/^[^{]+//' ${trace}
+    sort -t":" -k 3 ${trace} -o ${trace}
```

Comment:
Can we use the `preprocess_log` func here?

Reply:
Fixed